components/inputs/input-number.js

```diff
@@ -199,21 +210,23 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 			console.warn('d2l-input-number component requires label text');
 		}
 		this.addEventListener('d2l-localize-behavior-language-changed', () => {
+			this._descriptor = getNumberDescriptor();
 			if (this._formattedValue.length > 0) {
 				this._updateFormattedValue();
 			}
 		});
 	}
 
 	render() {
-		const tooltip = !this.disabled && this.validationError && this.childErrors.size === 0 ? html`<d2l-tooltip announced for="${this._inputId}" state="error" align="start">${this.validationError}</d2l-tooltip>` : null;
 		return html`
 			<d2l-input-text
 				autocomplete="${ifDefined(this.autocomplete)}"
 				?noValidate="${this.noValidate}"
 				?autofocus="${this.autofocus}"
 				@blur="${this._handleBlur}"
 				@change="${this._handleChange}"
+				@input="${this._handleInput}"
+				@keypress="${this._handleKeyPress}"
```

Comment:
Was there a reason to use `keypress` instead of `keydown`/`keyup`, given that [`keypress` is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event)?

Reply:
Mostly was following examples of suppressing characters, and `keydown` fires for everything (shift, tab, etc.) so separating out the things I cared about is more difficult with it. But yeah deprecated is no good so I'll try to fix this to work with `keydown`.

Reply:
Sounds good.  I think we do have some other stuff that relies on `keypress`.

Reply:
I looked into switching to `keydown` and... it's complicated. I got something kind-of working, but because it fires for ALL keys -- including Shift/Tab/Arrow/etc. -- it's much more difficult to only suppress "not numbers". I added a bunch of keys to let through, but then ran into cases where "Ctrl-Z" would be suppressed because the key is "z".

So after a conversation with Dave B we decided we'll stick with `keypress` for now, and swap it out for `beforeinput` once legacy-Edge goes away. `beforeinput` exposes `inputType` which lets us check for text insertion events more selectively.